fili-core/src/test/groovy/com/yahoo/bard/webservice/web/util/PaginationParametersSpec.groovy

```diff
@@ -130,7 +133,16 @@ class PaginationParametersSpec extends Specification {
         thrown(BadPaginationException)
 
         where:
-        invalidPerPage << ["0", "-1", "1.0", "2.3", "$Long.MAX_VALUE", "1a", "AndrewIsMostlyHarmless."]
+        invalidPerPage << ["0", "-2", "1.0", "2.3", "$Long.MAX_VALUE", "1a", "AndrewIsMostlyHarmless."]
+    }
+
+    def "If the result set is empty and the page is last then getPage returns the first page"() {
```

Comment:
You could combine this with `build the correct handler when 'perPage' is #perPage and 'page' is #page` if you make the `resultSize` a parameter